src/utils/ticket.js

```diff
@@ -0,0 +1,215 @@
+import axios from "axios";
+import { toast } from "react-toastify";
+import { BASE_URL } from "../actions/baseApi";
+
+export const createNewTicket = async function (newTicket) {
+  try {
+    const ticket = (
+      await axios.post(`${BASE_URL}/ticket`, newTicket, {
+        cancelToken: this.axiosCancel.token,
+      })
+    ).data.ticket;
+    ticket.comments = 0;
+    this.setState({
+      all: [ticket, ...this.state.all],
+      filtered: [ticket, ...this.state.all],
+      editorMode: false,
+      spinner: "",
+    });
+  } catch (err) {
+    console.log(err);
+    toast.error("Something went wrong! could create Ticket");
+    this.setState({ spinner: "" });
+  }
+};
+
+export const deleteTicket = async function (id) {
+  try {
+    const newTickets = this.state.all.filter((ele) => ele._id !== id);
+    await axios.delete(`${BASE_URL}/ticket/${id}`, {
+      cancelToken: this.axiosCancel.token,
+    });
+    this.setState({
+      all: [...newTickets],
+      filtered: [...newTickets],
+      spinner: "",
+    });
+  } catch (err) {
+    console.log(err);
+    toast.error("Something went wrong! could not delete Ticket");
+    this.setState({ spinner: "" });
+  }
+};
+
+export const saveTicketTitle = async function () {
+  const { title } = this.state;
+  await this.props.updateTicket({
+    type: "title",
+    title
+  });
+  this.props.singleUpdate(this.props.ticket._id, {
+    title
+  });
+};
+
+export const saveTicketSummary = async function () {
+  const { data } = this.state;
+  await this.props.updateTicket({
+    type: "shortDescription",
+    shortDescription: data,
+  });
+  this.props.singleUpdate(this.props.ticketId, {
+    shortDescription: data,
+  });
+};
+
+export const getTicket = async function () {
+  const ticket = (
+    await axios.get(`${BASE_URL}/ticket/${this.props.ticketId}`, {
+      cancelToken: this.axiosCancel.token,
+    })
+  ).data.ticket;
+  this.editsAllowed =
+    localStorage.getItem("ticketModerator") === "true" ||
```

Comment:
I think it's better to make it a function because this could be used in various component, which will follow DRY, KISS